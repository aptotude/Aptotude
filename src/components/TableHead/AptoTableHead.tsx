import React from 'react';
import { StandardTypes } from '../../utils/standardTypes';
import classNames from 'classnames';

const COMPONENT_PREFIX = 'AptoTableHead';

export class AptoTableHead extends React.PureComponent<StandardTypes> {
  public render() {
    const {
      children,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX
    );

    return (
      <thead className={classes} {...rest}>
        {children}
      </thead>
    );
  }
}
