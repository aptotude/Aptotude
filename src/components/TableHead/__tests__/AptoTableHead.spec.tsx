import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoTableHead } from '../AptoTableHead';

afterEach(() => {
  cleanup();
});

describe('AptoTableHead', () => {
  it('Renders Component', () => {
    const { container } = render(
      <table>
        <AptoTableHead>
          <tr>
            <th>heading</th>
          </tr>
        </AptoTableHead>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders extra class names', () => {
    const { container } = render(
      <table>
        <AptoTableHead className="foo">
          <tr>
            <th>heading</th>
          </tr>
        </AptoTableHead>
        <tbody>
          <tr>
            <td>hi</td>
          </tr>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
