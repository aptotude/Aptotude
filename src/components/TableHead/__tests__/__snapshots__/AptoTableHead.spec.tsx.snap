// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoTableHead Renders Component 1`] = `
<table>
  <thead
    class="AptoTableHead"
  >
    <tr>
      <th>
        heading
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableHead Renders extra class names 1`] = `
<table>
  <thead
    class="foo"
  >
    <tr>
      <th>
        heading
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;
